x-pack/test/cloud_security_posture_functional/pages/findings.ts

```diff
@@ -24,23 +28,112 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
+      it('add filter', async () => {
+        await filterBar.addFilter('rule.name', 'is', ruleName1);
+
+        expect(await filterBar.hasFilter('rule.name', ruleName1)).to.be(true);
+        expect(await table.hasColumnValue('Rule', ruleName1)).to.be(true);
+      });
+
+      it('remove filter', async () => {
+        await filterBar.removeFilter('rule.name');
+
+        expect(await filterBar.hasFilter('rule.name', ruleName1)).to.be(false);
+        expect(await table.getRowsCount()).to.be(data.length);
+      });
+
+      it('set search query', async () => {
+        await queryBar.setQuery(ruleName1);
+        await queryBar.submitQuery();
+
+        expect(await table.hasColumnValue('Rule', ruleName1)).to.be(true);
+        expect(await table.hasColumnValue('Rule', ruleName2)).to.be(false);
+
+        await queryBar.setQuery('');
+        await queryBar.submitQuery();
+
+        expect(await table.getRowsCount()).to.be(data.length);
+      });
     });
 
-    describe('Sort', () => {
-      it('Sorts by rule name', async () => {
-        await pageObjects.findings.table.toggleColumnSortOrFail('Rule', 'asc');
+    describe('Table Filters', () => {
+      it('add cell value filter', async () => {
+        await table.addCellFilter('Rule', ruleName1, false);
+
+        expect(await filterBar.hasFilter('rule.name', ruleName1)).to.be(true);
+        expect(await table.hasColumnValue('Rule', ruleName1)).to.be(true);
       });
 
-      it('Sorts by resource name', async () => {
-        await pageObjects.findings.table.toggleColumnSortOrFail('Resource Name', 'desc');
+      it('add negated cell value filter', async () => {
+        await table.addCellFilter('Rule', ruleName1, true);
+
+        expect(await filterBar.hasFilter('rule.name', ruleName1, true, false, true)).to.be(true);
+        expect(await table.hasColumnValue('Rule', ruleName1)).to.be(false);
+        expect(await table.hasColumnValue('Rule', ruleName2)).to.be(true);
+
+        await filterBar.removeFilter('rule.name');
+      });
+    });
+
+    describe('Table Sort', () => {
+      it('sorts by rule name', async () => {
+        await table.toggleColumnSortOrFail('Rule', 'asc');
+      });
+
+      it('sorts by resource name', async () => {
+        await table.toggleColumnSortOrFail('Resource Name', 'desc');
+      });
+    });
+
+    describe('DistributionBar', () => {
+      it('filters by passed findings', async () => {
+        await distributionBar.filterBy('passed');
+
+        const passed = data
+          .filter(({ result }) => result.evaluation === 'passed')
+          .map(() => 'Pass');
+
+        expect(await table.getColumnValues('Result')).to.eql(passed);
+
+        await filterBar.removeFilter('result.evaluation');
+      });
+
+      it('filters by failed findings', async () => {
+        await distributionBar.filterBy('failed');
+
+        const failed = data
+          .filter(({ result }) => result.evaluation === 'failed')
+          .map(() => 'Fail');
```

Comment:
it'd probably be better if the test didn't know we map `passed` to `pass` 

might change this later

Reply:
correct, instead consider tagging with `data-test-subj` if it is a failed result or passed
That way, you are not relaying on the visualization of the passed/failed